lib/gooddata/lcm/actions/synchronize_users.rb

```diff
@@ -247,24 +247,22 @@ def call(params)
                                            do_not_touch_users_that_are_not_mentioned: do_not_touch_users_that_are_not_mentioned,
                                            create_non_existing_user_groups: create_non_existing_user_groups)
                     when 'sync_multiple_projects_based_on_custom_id'
-                      new_users.group_by { |u| u[:pid] }.flat_map do |client_id, users|
+                      all_clients = domain.clients(:all, data_product).to_a
+                      new_users.group_by { |u| u[:pid] }.map do |client_id, n_users|
                         fail "Client id cannot be empty" if client_id.blank?
-                        begin
-                          project = domain.clients(client_id, data_product).project
-                        rescue RestClient::BadRequest => e
-                          raise e unless /does not exist in data product/ =~ e.response
-                          fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\""
-                        end
+                        c = all_clients.detect { |specific_client| specific_client.id == client_id }
+                        fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\"" if c.nil?
+                        project = c.project
                         fail "Client #{client_id} does not have project." unless project
-                        puts "Project #{project.pid} of client #{client_id} will receive #{users.count} users"
-                        project.import_users(users,
+                        puts "Project #{project.pid} of client #{client_id} will receive #{n_users.count} users"
+                        project.import_users(n_users,
```

Comment:
Similar code found in synchronize_users.rb:193, synchronize_users.rb:205, synchronize_users.rb:222, synchronize_users.rb:248, synchronize_users.rb:264, synchronize_users.rb:291, synchronize_users.rb:333 (mass = 112)